src/defs.h

```diff
@@ -45,6 +45,16 @@
 /* Number of the available registers. Either 7 or 8 is accepted now. */
 #define REG_CNT 8
 
+/* This macro will be automatically defined at shecc run-time. */
+#ifdef __SHECC__
+#define UNUSED(x) (x)
+#define SHECC_CAST(x) (x)
+#else
+/* suppress GCC/Clang warnings */
+#define UNUSED(x) (void) (x)
+#define SHECC_CAST(x) (void *) (x)
```

Comment:
I don't get the idea why `SHECC_CAST` exists. Can you explain?

Reply:
Is it necessary?

Reply:
> I don't get the idea why `SHECC_CAST` exists. Can you explain?

In printf/fprintf, GCC/Clang will check the format string specified with `-Wall` flag
```c
fprintf(fd, "subgraph cluster_%p {\n", SHECC_CAST(bb));
```

> Is it necessary?

Or we could simply apply the `-Wno-format` flag.

Reply:
> Or we could simply apply the `-Wno-format` flag.

Let's enforce `-Wno-format` option.